File: src/main/java/io/github/onepieceapi/InitProject.java
package io.github.onepieceapi;

import org.springframework.stereotype.Service;

import io.github.onepieceapi.constants.Urls;
import io.github.onepieceapi.model.Character;
import io.github.onepieceapi.repositories.CharacterRepository;
import jakarta.annotation.PostConstruct;
import lombok.AllArgsConstructor;
import lombok.extern.slf4j.Slf4j;

@Slf4j
@Service
@AllArgsConstructor
public class InitProject {
    private CharacterRepository characterRepository;

    @PostConstruct
    public void init() {
        log.info("Project initialized");
        Character luffy = new Character();
        luffy.setSurname("Monkey");
        luffy.setName("D. Luffy");

        Character zoro = new Character();
        zoro.setSurname("Roronoa");
        zoro.setName("Zoro");

        characterRepository.save(luffy);
        characterRepository.save(zoro);
    }
}

File: src/main/kotlin/io/github/onePieceApi/InitProject.kt
package io.github.onePieceApi

import io.github.onePieceApi.model.Chapter
import io.github.onePieceApi.model.Character
import io.github.onePieceApi.repositories.ChapterRepository
import io.github.onePieceApi.repositories.CharacterRepository
import jakarta.annotation.PostConstruct
import org.springframework.stereotype.Service

@Service
class InitProject constructor(
    private val characterRepository: CharacterRepository,
    private val chapterRepository: ChapterRepository
) {

    @PostConstruct
    fun init() {
        val luffy = Character()
        luffy.surname = "Monkey"
        luffy.name = "D. Luffy"
        characterRepository.save(luffy)

        createChapter("Romance Dawn - Amanecer de una aventura", 1)
        createChapter("El hombre del sombrero de paja, Luffy", 2)
        createChapter("Zoro, el cazador de recompensas", 3)
        createChapter("El capitán Morgan, \"Él del brazo de hacha\"", 4)
        createChapter("El rey de los piratas", 5)
        createChapter("El primero", 6)
        createChapter("Amigos", 7)
        createChapter("Aparece Nami", 8)
    }

    private fun createChapter(title: String, number: Int) {
        val chapter = Chapter()
        chapter.title = title
        chapter.number = number
        chapterRepository.save(chapter)
    }
}
